src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl.py

```diff
@@ -866,33 +880,29 @@ def __call__(
         with self.progress_bar(total=num_inference_steps) as progress_bar:
             for i, t in enumerate(timesteps):
                 # expand the latents if we are doing classifier free guidance
-                latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
-                latent_model_input = self.scheduler.scale_model_input(latent_model_input, t)
+                scaled_latents = self.scheduler.scale_model_input(latents, t)
+                latent_model_input = torch.cat([scaled_latents] * 2) if do_classifier_free_guidance else scaled_latents
```

Comment:
Could I have an explanation on why this differs from 

https://github.com/huggingface/diffusers/blob/6b1abba18dce2eb169b6363ea5f626e7cd87cf21/src/diffusers/pipelines/controlnet/pipeline_controlnet.py#L924 

?

Reply:
Just wanted to make the code around switching CFG more clear, I don't see the point behind scaling latents once for main unet and then the second time for controlnet. So I made a change which is aligned with all inputs (`prompt embeds`, `text_embeds`, `time_ids`): we first prepare the standard inputs for controlnet, and then expand them (`torch.cat([...]*2)` for unet if it is required by CFG

Reply:
Got it. Clean.

Reply:
I just took a closer look here. Can we please just do exactly the same as here? https://github.com/huggingface/diffusers/blob/80871ac5971fe7e708befa3b553463c4e61b22ab/src/diffusers/pipelines/controlnet/pipeline_controlnet.py#L938C19-L938C19

I think the logic is very clear there:
1. all the code create inputs for the `controlnet_model` is addressed within the  `if ... else ...` statement
2. it is very clear to me that only when `guess_mode and do_classifier_free_guidance` it will differ from our regular Unet model input